app/src/main/java/com/mobileasone/dagger2workshop/di/RepositoryFactory.kt

```diff
@@ -0,0 +1,16 @@
+package com.mobileasone.dagger2workshop.di
+
+import com.mobileasone.dagger2workshop.data.local.NotesLocalMemoryRepository
+import com.mobileasone.dagger2workshop.domain.repositories.NotesRepository
+
+object RepositoryFactory {
+
+    private val notesLocalMemoryRepositoryInstance: NotesRepository by lazy { initNotesRepository() }
```

Comment:
By doing this, we have a single instance of NotesRepository like we had with the getInstance method in [NotesLocalMemoryRepository.kt](https://github.com/Jamargle/Dagger2Workshop/pull/3/files#diff-f4a68c69132466c2d7cffd85fed4bd62)